src/components/SearchForm/SearchForm.js

```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+import { useQueryParam, StringParam } from 'use-query-params';
+import {
+  SearchFormWrapper,
+  Title,
+  Form,
+  Prefix,
+  Input,
+} from './SearchForm.styles';
+
+import { CTAButton } from '../CTAButton/CTAButton.styles';
+
+const SearchForm = () => {
+  const [subreddit, setSubreddit] = useQueryParam('q', StringParam);
+
+  return (
+    <SearchFormWrapper>
+      <Title>Find the best time for a subreddit</Title>
+      <Form>
+        <Prefix>r /</Prefix>
+        <Input type="text" value={subreddit} onChange={(e) => setSubreddit(e.target.value)} />
```

Comment:
Fyi: a lot of people don't know this so I'm mentioning it here. Assigning callback like this

```
<Input onChange={(e) => setSubreddit(e.target.value)} />
// or
const onChange = (e) => setSubreddit(e.target.value);
return <Input onChange={onChange} />
```

will force the child component (here Input) to re-render every time when the parent (here SearchForm) is re-rendered. This is because the callback is newly created on every render. Thus the reference changes and the child component needs to re-render as well.

That's not a problem here, because Input is just a native `input` element. But this can lead to nasty bugs when you pass down callbacks like this to more complicated components. You have the same problem when you do this

```
<MyComponent myProp={someObject || {}} />
```

You can see that from time to time but it's dangerous for the same reason

Reply:
What would be a better approach to not cause re-renders?

Reply:
Should have mentioned that as well, right :-). `useCallback` is the way to go. But not necessary here